src/js/jsx/search/SearchBar.jsx

```diff
@@ -500,20 +500,29 @@ define(function (require, exports, module) {
 
         render: function () {
             var searchStrings = nls.localize("strings.SEARCH"),
-                noMatchesOption = {
+                filters = this.state.filter,
+                placeholderText,
+                clearInputBtn;
+
+            var flux = this.getFlux(),
+                searchStore = flux.store("search"),
+                registeredSearchModules = searchStore._registeredSearchTypes;
+
+            var lastFilter = filters[filters.length - 1];
+
+            var searchPayload = registeredSearchModules[lastFilter];
+            var noOptionsString = searchPayload ? searchPayload.noOptionsString : null;
+            var defaultNoOptions = nls.localize("strings.SEARCH.NO_OPTIONS");
```

Comment:
In addition to comment above, instead of doing this in `render`, can you move all these calculations to `getStateFromFlux` method so the no options string to be shown is part of the state?